src/main/java/pl/javastart/task/Room.java

```diff
@@ -0,0 +1,45 @@
+package pl.javastart.task;
+
+public class Room {
+    private final double area;
+    private double currentTemperature;
+    private final boolean hasAirConditioner;
+    private final double temperatureLowerLimit;
+
+    public Room(double area, double currentTemperature, boolean hasAirConditioner, double temperatureLowerLimit) {
+        this.area = area;
+        this.currentTemperature = currentTemperature;
+        this.hasAirConditioner = hasAirConditioner;
+        this.temperatureLowerLimit = temperatureLowerLimit;
+    }
+
+    public boolean decreaseTemperature() {
+        if (!hasAirConditioner || currentTemperature == temperatureLowerLimit) {
+            return false;
+        }
+
+        currentTemperature -= 1;
+
+        if (currentTemperature < temperatureLowerLimit) {
+            currentTemperature = temperatureLowerLimit;
+        }
```

Comment:
O ile tutaj to nie ma większego znaczenia, to nie jest najlepszym pomysłem na chwilę zmienić wartość pola poniżej dozwolonego, żeby za chwilę przywrócić do wartości minimalnej. Lepiej stworzyć w tym celu zmienną, sprawdzić, czy jej wartość jest w porządku, a jeśli tak to podstawić pod pole. Unikniesz w takim podejściu sytuacji, gdzie przez moment system nie spełnia nałożonych wymagań.

Reply:
Słuszna uwaga. Będę na przyszłość pamiętać, by tego błędu unikać.